src/nemos/utils.py

```diff
@@ -248,120 +254,268 @@ def _pad_dimension(
+
+        1. Single array of shape `(n_trials, n_time_bins, n_neurons,
+           n_features)`
+        2. Pytree whose leaves are arrays of shape `(n_time_bins, n_neurons,
+           n_features)`
+    pad_size:
+        The number of NaNs to concatenate as padding.
+    predictor_causality:
+        Causality of this predictor, which determines where padded values are added.
 
     Returns
     -------
-    :
-        A pytree with the padded trials that has the same structure as conv_trials.
+    padded_conv_time_series :
+        `conv_time_series` with NaN padding. NaN location is determined by
+        value of `predictor_causality` (see Notes), and `padded_conv_time_series`
+        structure is determined by that of `conv_time_series`:
+
+        1. Single array of shape `(n_trials, n_time_bins + pad_size,
+           n_neurons, n_features)`
+        2. Pytree whose leaves are arrays of shape `(n_time_bins + pad_size,
+           n_neurons, n_features)`
 
     Raises
     ------
     ValueError
-        If the window_size is not a positive integer, or if the filter_type is not one of 'causal',
-        'acausal', or 'anti-causal'. Also raises ValueError if the dimensionality of conv_trials is not as expected.
+        - If conv_time_series does not have a float dtype.
+        - If pad_size is not a positive integer
+        - If predictor_causality is not one of 'causal', 'acausal', or 'anti-causal'.
+        - If the dimensionality of conv_trials is not as expected.
+    warning
+        - If pad_size is odd and predictor_causality=='acausal'. In order for
+          the output to be truly acausal, i.e., symmetric around events found
+          in `conv_trials`, we need to be able to add an equal number of NaNs
+          on both sides.
 
     Notes
     -----
-    The `adjust_indices` dictionary specifies the start and end indices for
-    slicing the input data based on the filter type.
+    The location of the NaN-padding depends on the value of `predictor_causality`.
+
+    - `'causal'`: `pad_size-1` NaNs are placed at the beginning of the
+      temporal dimension.
+    - `'acausal'`: `floor(pad_size/2)` NaNs are placed at the beginning of
+      the temporal dimensions, `ceil(pad_size/2)` placed at the end.
+    - `'anti-causal'`: `pad_size-1` NaNs are placed at the end of the
+      temporal dimension.
+
+    """
+    if not isinstance(pad_size, int) or pad_size <= 0:
+        raise ValueError(
+            f"pad_size must be a positive integer! Pad size of {pad_size} provided instead!"
+        )
+
+    causality_choices = ["causal", "acausal", "anti-causal"]
+    if predictor_causality not in causality_choices:
+        raise ValueError(
+            f"predictor_causality must be one of {causality_choices}. {predictor_causality} provided instead!"
+        )
+    if predictor_causality == "acausal" and (pad_size % 2 == 0):
+        warnings.warn(
+            "With acausal filter, pad_size should probably be even,"
```

Comment:
Should catch this warning in `create_convolutional_predictor` and reword it so it matches the arguments there (`basis_matrix.shape[0] should be odd` or similar)

Reply:
note that it's both the argument name and even -> odd that need to change